cmd/kubeadm/app/apis/output/v1alpha1/types.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeadmapiv1beta2 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta2"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BootstrapToken represents information for the bootstrap token output produced by kubeadm
+// This is a copy of BoostrapToken struct from ../kubeadm/types.go with 2 additions:
+// metav1.TypeMeta and metav1.ObjectMeta
+type BootstrapToken struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:",inline"`
```

Comment:
@rosti you had objections about adding the full metav1 types to our objects, because we don't need a lot of the automatic/dynamic fields that the api server manages.

yet, the structured output needs them.
maybe we should still include the full ObjectMeta there for consistency instead of defining a subset.

@bart0sh 
i remember we were in doubt whether we should duplicate BoostrapToken under the output group.
we may still wan to do that, but wondering whether 
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta2/types.go#L228
should have TypeMeta and ObjectMeta
and this structure should look like:

```
type BootstrapToken struct {
    kubeadmapiv1beta2.BootstrapToken
}
```

Reply:
I'd prefer not to modify kubeadm API just for this. Another point is that  kubeadmapiv1beta2.BootstrapToken doesn't need TypeMeta and ObjectMeta fields as they're only required by output machinery, hence it's better to have them only in the output API in my opinion.